my-app/src/Components/Auth.tsx

```diff
@@ -1,33 +1,37 @@
-import React from "react";
+import React, { useState } from "react";
 import styles from "../css/Auth.module.css";
 import { Title } from "./Title";
 import { Header } from "./Header";
 
 type Props = {
-   onAuth: (login: string, pass: string) => void
+   onAuth: (login: string, pass: string) => void,
+   onRegClick: () => void
 }
+export const Auth = ({ onAuth, onRegClick }: Props) => {
```

Comment:
auth and reg are different components, even if they look very similar.
prefer to separate them. In the beginning - you can do it in the same file